numpy/core/src/multiarray/arrayobject.c

```diff
@@ -455,7 +455,9 @@ array_dealloc(PyArrayObject *self)
     if ((fa->flags & NPY_ARRAY_OWNDATA) && fa->data) {
         /* Free any internal references */
         if (PyDataType_REFCHK(fa->descr)) {
-            PyArray_ClearArray(self);
+            if (PyArray_ClearArray(self) < 0) {
```

Comment:
Just to check I understand: to write a test for this, one would have to have a clearing function that actually misbehaves, correct? And the actual array object here is no longer accessible, hence `NULL`?

Reply:
Yes, it shouldn't be possible to reach this, but without the fix, you do.